import_flow_events.py

```diff
@@ -84,18 +130,81 @@
     CREDENTIALS 'aws_access_key_id={aws_access_key_id};aws_secret_access_key={aws_secret_access_key}'
     FORMAT AS CSV;
 """
-
-Q_CHECK_FOR_DAY = """
-    SELECT timestamp FROM flow_events
-    WHERE timestamp >= {timestamp}
-    AND timestamp < {timestamp} + 86400
-    LIMIT 1;
+Q_INSERT_METADATA = """
+    WITH durations AS (
+      SELECT flowId, MAX(flowTime)
+      FROM flow_csv
+      GROUP BY flowId
+    )
+    INSERT INTO flow_metadata (
+      flowId,
+      beginTime,
+      duration,
+      completed,
+      newAccount,
+      browser,
+      browserVersion,
+      os,
+      context,
+      entrypoint,
+      migration,
+      service,
+      utmCampaign,
+      utmContents,
+      utmMedium,
+      utmSource,
+      utmTerm
+    )
+    SELECT (
+      begin.flowId,
+      (begin.timestamp * 1000.0),
+      (durations.flowTime * 1000.0),
+      (CASE WHEN signed.flowId IS NULL THEN FALSE ELSE TRUE END),
+      (CASE WHEN created.flowId IS NULL THEN FALSE ELSE TRUE END),
+      begin.browser,
+      begin.browserVersion,
+      begin.os,
+      begin.context,
+      begin.entrypoint,
+      begin.migration,
+      begin.service,
+      begin.utmCampaign,
+      begin.utmContents,
+      begin.utmMedium,
+      begin.utmSource,
+      begin.utmTerm
+    )
+    FROM flow_csv AS begin
+    INNER JOIN durations
+      ON begin.flowId = durations.flowId AND begin.type = 'flow.begin'
+    LEFT JOIN flow_csv AS created
+      ON begin.flowId = created.flowId AND created.type = 'account.created'
+    LEFT JOIN flow_csv AS signed
+      ON begin.flowId = signed.flowId AND signed.type = 'account.signed';
```

Comment:
> This might make it easier to handle flows that cross the boundary between days

Excellent point, I hadn't thought that scenario through properly at all. Will change to updates.